src/jit/lower.cpp

```diff
@@ -3408,6 +3408,9 @@ bool Lowering::IndirsAreEquivalent(GenTreePtr candidate, GenTreePtr storeInd)
     if (pTreeA->OperGet() != pTreeB->OperGet())
         return false;
 
+    if (candidate->gtType != storeInd->gtType)
```

Comment:
I still can't reproduce the byte issue, before lowering both indirection types are ubyte:

>
```
( 11, 13) [000007] ------------             *  stmtExpr  void  (top level) (IL 0x000...0x009)
N001 (  4,  5) [000005] ----G--N----             |  /--*  &clsVar   byref  Hnd=0x8c1a5cc0 $200
N002 (  4,  5) [000001] ----G-------             |  |     /--*  &clsVar   byref  Hnd=0x8c1a5cc0 <l:$181, c:$180>
     (  4,  5) [000013] ----G-------             |  |  /--*  indir     ubyte
N003 (  1,  1) [000002] ------------             |  |  +--*  const     int    16 $42
N004 (  6,  7) [000003] ----G-------             |  +--*  |         int    <l:$183, c:$182>
N005 ( 11, 13) [000006] -A--G-------             \--*  storeIndir ubyte  <l:$183, c:$182>
```